WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/builders/ThreatItemBuilder.kt

```diff
@@ -0,0 +1,93 @@
+package org.wordpress.android.ui.jetpack.scan.builders
+
+import dagger.Reusable
+import org.wordpress.android.R
+import org.wordpress.android.fluxc.model.scan.threat.ThreatModel
+import org.wordpress.android.fluxc.model.scan.threat.ThreatModel.CoreFileModificationThreatModel
+import org.wordpress.android.fluxc.model.scan.threat.ThreatModel.DatabaseThreatModel
+import org.wordpress.android.fluxc.model.scan.threat.ThreatModel.FileThreatModel
+import org.wordpress.android.fluxc.model.scan.threat.ThreatModel.GenericThreatModel
+import org.wordpress.android.fluxc.model.scan.threat.ThreatModel.VulnerableExtensionThreatModel
+import org.wordpress.android.fluxc.model.scan.threat.ThreatModel.VulnerableExtensionThreatModel.Extension.ExtensionType
+import org.wordpress.android.ui.jetpack.scan.ScanListItemState.ThreatItemState
+import org.wordpress.android.ui.utils.UiString.UiStringRes
+import org.wordpress.android.ui.utils.UiString.UiStringResWithParams
+import org.wordpress.android.ui.utils.UiString.UiStringText
+import javax.inject.Inject
+
+@Reusable
+class ThreatItemBuilder @Inject constructor() {
+    fun buildThreatItem(threatModel: ThreatModel, onThreatItemClicked: (threatModel: ThreatModel) -> Unit) =
+        ThreatItemState(
+            threatId = threatModel.baseThreatModel.id,
+            header = buildThreatItemHeader(threatModel),
+            subHeader = buildThreatItemSubHeader(threatModel),
+            onClick = { onThreatItemClicked(threatModel) }
+        )
+
+    private fun buildThreatItemHeader(threatModel: ThreatModel) = when (threatModel) {
+        is CoreFileModificationThreatModel -> UiStringResWithParams(
+            R.string.threat_item_header_infected_core_file,
+            listOf(UiStringText(getDisplayFileName(threatModel.fileName)))
+        )
+
+        is DatabaseThreatModel -> UiStringResWithParams(
+            R.string.threat_item_header_database_threat,
+            listOf(UiStringText("${threatModel.rows?.size ?: 0}"))
+        )
+
+        is FileThreatModel -> UiStringResWithParams(
+            R.string.threat_item_header_file_malicious_code_pattern,
+            listOf(UiStringText(getDisplayFileName(threatModel.fileName)))
+        )
+
+        is VulnerableExtensionThreatModel -> {
+            val slug = threatModel.extension.slug ?: ""
+            val version = threatModel.extension.version ?: ""
+            when (threatModel.extension.type) {
+                ExtensionType.PLUGIN -> UiStringResWithParams(
+                    R.string.threat_item_header_vulnerable_plugin,
+                    listOf(UiStringText(slug), UiStringText(version))
+                )
+                ExtensionType.THEME -> UiStringResWithParams(
+                    R.string.threat_item_header_vulnerable_theme,
+                    listOf(UiStringText(slug), UiStringText(version))
+                )
+                ExtensionType.UNKNOWN -> throw IllegalArgumentException(
```

Comment:
⚠️ I'm not sure this is a case in which we should use Fail Fast principle. AFAIU the data are coming from the server. If the server introduces a new ExtensionType I'm not sure if our app (especially old versions of the app) should start crashing. Introduction of a new ExtensionType is not a breaking change from the API point of view. I'd consider wrapping the exception into "if (BuildConfig.DEBUG)" and log a message in production. Are my assumptions correct? Wdyt?

Reply:
I filtered out `ExtensionType.UNKNOWN` types in fluxc [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/develop/fluxc/src/main/java/org/wordpress/android/fluxc/model/scan/threat/ThreatMapper.kt#L43) so they're returned as `GenericThreatModel` as we might not know how this unknown type should be rendered. Ideally this condition will not be reached.

As suggested, I'll also add this check and a log for production with an appropriate tag once [this PR ](https://github.com/wordpress-mobile/WordPress-Utils-Android/pull/59) to add `AppLog` tags for the jetpack features in `WordPress-Utils-Android` is merged.

Reply:
Ohh I see, thanks for the clarification. In that case I don't mind crashing - I'm a fan of fail fast principle. :)